core/domain/interaction_registry.py

```diff
@@ -143,3 +143,44 @@ def get_all_specs_for_state_schema_version(cls, state_schema_version):
 
         return cls._state_schema_version_to_interaction_specs[
             state_schema_version]
+
+    @classmethod
+    def get_all_specs_for_state_schema_version_or_latest(
+        cls,
+        state_schema_version,
+    ):
+        """Returns a dict containing the full specs of each interaction for the
+        given state schema version, if available else return the latest specs.
+
+        Args:
+            state_schema_version: int. The state schema version to retrieve
+                interaction specs for.
+
+        Returns:
+            dict. The interaction specs for the given state schema
+            version, in the form of a mapping of interaction id to the
+            interaction specs. See interaction_specs.json for an example.
+
+        Raises:
+            Exception. No interaction specs json file found for the given state
+                schema version.
+        """
+        if (state_schema_version not in
+                cls._state_schema_version_to_interaction_specs):
+            file_name = (
+                'interaction_specs_state_v%i.json' % state_schema_version)
+            spec_file = os.path.join(
+                feconf.INTERACTIONS_LEGACY_SPECS_FILE_DIR, file_name)
+
+            if os.path.isfile(spec_file):
+                with python_utils.open_file(spec_file, 'r') as f:
+                    specs_from_json = json.loads(f.read())
+                cls._state_schema_version_to_interaction_specs[
+                    state_schema_version] = specs_from_json
+                return cls._state_schema_version_to_interaction_specs[
+                    state_schema_version]
+            else:
+                return cls.get_all_specs()
+
+        return cls._state_schema_version_to_interaction_specs[
+            state_schema_version]
```

Comment:
Most of the part of this function looks like a duplicate of the above function `get_all_specs_for_state_schema_version`, maybe change the above function to have keywordParam to fetch latest? OR use the `get_all_specs_for_state_schema_version` function here?

Reply:
Thanks @DubeySandeep for the feedback, I will address this in my next PR.